habitat/tasks/rearrange/rearrange_sensors.py

```diff
@@ -0,0 +1,618 @@
+            high=np.finfo(np.float32).max,
+            dtype=np.float32,
+        )
+
+
+@registry.register_sensor
+class TargetObjectSensor(MultiObjSensor):
+    """
+    Relative to the end effector. This is the ground truth position, accurate
+    at every time step.
+    """
+
+    cls_uuid: str = "obj_cur_sensor"
+
+    def get_observation(self, observations, episode, *args, **kwargs):
+        idxs, _ = self._sim.get_targets()
+        scene_pos = self._sim.get_scene_pos()
+        pos = scene_pos[idxs]
+
+        ee_pos = self._sim.robot.ee_transform.translation
+        to_targ = pos - ee_pos
+        trans = self._sim.get_robot_transform()
+        for i in range(to_targ.shape[0]):
+            to_targ[i] = trans.inverted().transform_vector(to_targ[i])
+        return to_targ
+
+
+@registry.register_sensor
+class AbsTargetObjectSensor(MultiObjSensor):
+    """
+    This is the ground truth position
+    """
+
+    cls_uuid: str = "abs_obj_cur_sensor"
+
+    def get_observation(self, observations, episode, *args, **kwargs):
+        idxs, _ = self._sim.get_targets()
+        scene_pos = self._sim.get_scene_pos()
+        pos = scene_pos[idxs]
+
+        return pos
+
+
+@registry.register_sensor
+class TargetStartSensor(MultiObjSensor):
+    """
+    Relative to the end effector
+    """
+
+    cls_uuid: str = "obj_start_sensor"
+
+    def get_observation(self, observations, episode, *args, **kwargs):
+        global_T = self._sim.robot.ee_transform
+        T_inv = global_T.inverted()
+        pos = self._sim.get_target_objs_start()
+        for i in range(pos.shape[0]):
+            pos[i] = T_inv.transform_point(pos[i])
+
+        return pos
+
+        # pos = self._sim.get_target_objs_start()
+        # ee_pos = self._sim.robot.ee_transform.translation
+        # to_targ = pos - ee_pos
+        # trans = self._sim.get_robot_transform()
+        # for i in range(to_targ.shape[0]):
+        #    to_targ[i] = trans.inverted().transform_vector(to_targ[i])
+        # return to_targ
+
+
+@registry.register_sensor
+class AbsTargetStartSensor(MultiObjSensor):
+    cls_uuid: str = "abs_obj_start_sensor"
+
+    def _get_observation_space(self, *args, **kwargs):
+        n_targets = self._sim.get_n_targets()
+        return spaces.Box(
+            shape=(n_targets, 3),
+            low=np.finfo(np.float32).min,
+            high=np.finfo(np.float32).max,
+            dtype=np.float32,
+        )
+
+    def get_observation(self, observations, episode, *args, **kwargs):
+        pos = self._sim.get_target_objs_start()
+        return pos
+
+
+@registry.register_sensor
+class DynObjPosStartOrGoal(MultiObjSensor):
+    """
+    Returns the goal position if the robot is holding an object and the object
+    starting position if the robot is not holding the object.
+    """
+
+    cls_uuid: str = "dyn_obj_start_or_goal_sensor"
+
+    def get_observation(self, observations, episode, *args, **kwargs):
+        global_T = self._sim.robot.ee_transform
```

Comment:
Based on the way this is being used, I think this is the `transform_world_ee` as it transforms a point from the `ee` coordinate frame to world coordinates.  Then `T_inv` is `transform_ee_world`.  Assuming that is correct, please rename